ferminet/train.py

```diff
@@ -290,11 +290,8 @@ def train(cfg: ml_collections.ConfigDict):
 
   # Initialisation done. We now want to have different PRNG streams on each
   # device. Shard the key over devices
-  key, subkeys = jax.random.split(key, num_devices+1)
+  key, *subkeys = jax.random.split(key, num_devices+1)
   subkeys = jnp.stack(subkeys)
-  # Handle num_devices=1 case by explicitly broadcasting from an array of shape
-  # (2,) to an array of (num_devices, 2). This is a no-op for num_devices > 1.
-  subkeys = jnp.broadcast_to(subkeys, (num_devices, 2))
```

Comment:
Seems not necessary for `num_devices == 1` any more because `subkeys` is also a list of jax array in that case now